bokeh/models/plots.py

```diff
@@ -240,6 +240,23 @@ def add_tile(self, tile_source, **kw):
         self.renderers.append(tile_renderer)
         return tile_renderer
 
+    def add_dynamic_image(self, image_source, **kw):
```

Comment:
I don't think `p.renderers.append(image_layer)` is right - it means you have to know an implementation detail. Either something is doable or not (it would be insconsistent if there was one renderer you couldn't add to plot via the api). Can DynamicImage be a glyph?

@mattpap, I'd love to learn more about you not liking `plot.add_tile`. Tiles seem to me like a different kind of thing with their own distinct behaviors/needs. Where were you thinking they should go? (Just trying to learn here).